### DISCUSSION TITLE
Running code on program termination

### DISCUSSION THREAD

**Discussion by olliejohnson** [cc-tweaked/CC-Tweaked#1783] [Q&A] [answered]
Is there a way to run a function when the program is terminated

**Comment by Lupus590** [comment, marked as answer]
I would wrap my code in a call and watch for the terminated error message.
I think I have a code snippet for this somewhere, I'll have a look for it.

Edit: [This snippet](https://github.com/Lupus590-CC/CC-Random-Code/blob/master/src/cleanupOnError.lua) allows for tidy-up when main function errors. You can inspect the err message to see if it matches the terminated error message and thus do some special action on terminate.

**Comment by Lupus590** [comment]
An alternative way would be to use [os.pullEventRaw](https://tweaked.cc/module/os.html#v:pullEventRaw) and thus not have the automatic termination error throwing. Your code will have to handle the terminate request event itself.
